### ISSUE TITLE
Does inference during model evaluation optimize activities?

### ISSUE THREAD

**Issue by kiranchari** [Bogacz-Group/PredictiveCoding#4]
Hi,

I have a question regarding the test function in 1_supervised_learning_pc.ipynb 

I notice that in eval mode (when self.training is False), PC layer simply outputs the input mu: https://github.com/Bogacz-Group/PredictiveCoding/blob/334a87eeb6c6cad5197de89453c2c0a9ff43a18a/predictive_coding/pc_layer.py#L307

So it appears that during model evaluation, the predicted activities are used to compute the accuracy rather than the activities that minimize the energy. Is this correct? If yes, may I know why the activities are not optimized during evaluation as in the standard PC setting?

Thanks!

**Comment by gaspardol** [comment]
Hi,

Thank you for taking the time to look into our codebase and for your thoughtful question.

You are correct, the model directly uses the predicted activities during evaluation. This is because, in the absence of a label, the predicted activities correspond to the states with minimal energy. Therefore, performing iterative energy minimization during evaluation would not change the result but would significantly increase computation time. Using the predicted activities instead provides the same outcome more efficiently.

Best,
Gaspard
